src/js/chartAuthoring.js

```diff
@@ -185,9 +232,59 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             source: "{that}.options.dataEntryPanel",
             removeSource: true,
             target: "{that dataEntryPanel}.options"
+        },
+        {
+            source: "{that}.options.inlineEditCommonOptions",
+            removeSource: true,
+            target: "{that fluid.inlineEdit}.options"
         }]
     });
 
+    // Adds and removes highlights as data plays
+    // uses floe.d3.filterById and floe.d3.filterByNotId
+    floe.chartAuthoring.highlightPlayingData = function(that) {
+        var currentlyPlayingData = that.model.currentlyPlayingData;
+        var chartAuthoringInterface = that.chartAuthoringInterface;
+
+        // The chart authoring interface is ready
+        if(chartAuthoringInterface !== undefined) {
+            var legendTable, tbody, rows, pie, slices;
+            var dataPlayingHighlightClass = that.options.styles.dataPlayingHighlightClass;
+            legendTable = chartAuthoringInterface.pieChart.legend.table;
+
+            pie = chartAuthoringInterface.pieChart.pie;
+            slices = pie.paths;
+
+            tbody = legendTable.selectAll("tbody");
+            rows = tbody.selectAll("tr");
+
+            // Nothing is currently playing; remove any highlighting
+            if(currentlyPlayingData === null) {
+                rows.classed(dataPlayingHighlightClass,false);
+                slices.classed(dataPlayingHighlightClass,false);
+            }
+
+            // Highlight pie chart slices / legend rows as they play
+            if(currentlyPlayingData !== null) {
+
+                var currentlyPlayingDataId = currentlyPlayingData.id;
```

Comment:
All of this code should be modelised rather than banging on the view layer directly